packages/use-dataloader/src/useDataLoader.js

```diff
@@ -0,0 +1,164 @@
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useReducer,
+  useRef,
+} from 'react'
+import { useDataLoaderContext } from './DataLoaderProvider'
+import { ActionEnum, StatusEnum } from './constants'
+import reducer from './reducer'
+
+const Actions = {
+  createReset: ({ data }) => ({ type: ActionEnum.RESET, data }),
+  createOnLoading: () => ({ type: ActionEnum.ON_LOADING }),
+  createOnSuccess: data => ({ type: ActionEnum.ON_SUCCESS, data }),
+  createOnUpdateData: data => ({ type: ActionEnum.ON_UPDATE_DATA, data }),
+  createOnError: error => ({ type: ActionEnum.ON_ERROR, error }),
+}
+
+const useDataLoader = (
+  key,
+  method,
+  {
+    onSuccess,
+    onError,
+    initialData,
+    pollingInterval,
+    enabled = true,
+    reloadOnKeyChange = true,
+    keepOldData = true,
```

Comment:
keepPreviousData or more implicite keepCachedData but this should be always true if we have key or cacheKey ?